crates/iota/src/client_commands.rs

```diff
@@ -1526,15 +1526,40 @@ impl IotaClientCommands {
                 match (amounts.as_ref(), count) {
                     (None, None) => bail!("You must use one of amounts or count options."),
                     (Some(_), Some(_)) => bail!("Cannot specify both amounts and count."),
-                    (None, Some(0)) => bail!("Coin split count must be greater than 0"),
+                    (None, Some(0)) | (None, Some(1)) => {
+                        bail!("Coin split count must be greater than 1")
+                    }
                     _ => { /*no_op*/ }
                 }
                 let client = context.get_client().await?;
-                let tx_kind = client
-                    .transaction_builder()
-                    .split_coin_tx_kind(coin_id, amounts, count)
-                    .await?;
                 let signer = context.get_object_owner(&coin_id).await?;
+                let gas_coins_page = client
+                    .coin_read_api()
+                    .get_coins(signer, None, None, None)
+                    .await?;
+                // If we only have a single coin, we have to split from the gas coin
+                let tx_kind = if gas_coins_page.data.len() == 1 {
+                    if let Some(amounts) = amounts {
+                        client
+                            .transaction_builder()
+                            .pay_iota_tx_kind(vec![signer; amounts.len()], amounts)?
+                    } else {
+                        let count_to_compute = count.expect("count is None");
```

Comment:
not the best error

Reply:
Any recommendation? Right now it's safe to unwrap here as we check above that the count must be defined if the amount is None

Reply:
In such cases I'd rather unwrap with a comment than expect, but not super opinionated

Reply:
Maybe construct an Either type above? Or construct the two vectors that get passed to `pay_iota_tx_kind` in the match so that we don't need the branch here.

Reply:
I changed it here https://github.com/iotaledger/iota/pull/6647/commits/e804d1becccfed4df0efdff91200bffc94bc99e5 this seems to be the cleanest solution to me